classes/ActionScheduler_AdminView.php

```diff
@@ -110,6 +110,89 @@ protected function get_list_table() {
 		return $this->list_table;
 	}
 
+	/**
+	 * Action: admin_notices
+	 *
+	 * Maybe check past-due actions, and print notice.
+	 *
+	 * @uses $this->check_pastdue_actions()
+	 */
+	public function maybe_check_pastdue_actions() {
+
+		# Filter to prevent checking actions (ex: inappropriate user).
+		if ( ! apply_filters( 'action_scheduler_check_pastdue_actions', true ) )
+			return;
+
+		# Get last check transient.
+		$last_check = get_transient( 'action_scheduler_last_pastdue_actions_check' );
+
+		# If transient exists, we're within interval, so bail.
+		if ( !empty( $last_check ) )
+			return;
+
+		# Perform the check.
+		$this->check_pastdue_actions();
+	}
+
+	/**
+	 * Check past-due actions, and print notice.
+	 *
+	 * @todo update $link_url to "Past-due" filter when released (see issue #510, PR #511)
+	 */
+	protected function check_pastdue_actions() {
+
+		# Set thresholds.
+		$threshold_seconds = ( int ) apply_filters( 'action_scheduler_pastdue_actions_seconds', DAY_IN_SECONDS );
+		$threshhold_min    = ( int ) apply_filters( 'action_scheduler_pastdue_actions_min', 1 );
+
+		# Allow third-parties to preempt the default check logic.
+		$check = apply_filters( 'action_scheduler_pastdue_actions_check_pre', null );
+
+		# Scheduled actions query arguments.
+		$query_args = array(
+			'date'     => as_get_datetime_object( time() - $threshold_seconds ),
+			'status'   => ActionScheduler_Store::STATUS_PENDING,
+			'per_page' => $threshhold_min,
+		);
+
+		# If no third-party preempted, run default check.
+		if ( is_null( $check ) ) {
+			$store = ActionScheduler_Store::instance();
+			$num_pastdue_actions = ( int ) $store->query_actions( $query_args, 'count' );
+
+			# Check if past-due actions count is greater than or equal to threshold.
+			$check = ( $num_pastdue_actions >= $threshhold_min );
+			$check = ( bool ) apply_filters( 'action_scheduler_pastdue_actions_check', $check, $num_pastdue_actions, $threshold_seconds, $threshhold_min );
+		}
+
+		# If check failed, set transient and abort.
+		if ( ! boolval( $check ) ) {
+			$interval = apply_filters( 'action_scheduler_pastdue_actions_check_interval', round( $threshold_seconds / 4 ), $threshold_seconds );
+			set_transient( 'action_scheduler_last_pastdue_actions_check', time(), $interval );
```

Comment:
@crstauf Is this transient used anywhere?

Reply:
@rrennick Yes, here:
https://github.com/woocommerce/action-scheduler/blob/8df980217252f37e510c3051068951de27c131d0/classes/ActionScheduler_AdminView.php#L120-L135

Reply:
Although it occurs to me now that if a site is using object caching, a transient may not survive the eviction algorithm. Thoughts?

Reply:
@rrennick 

> Although it occurs to me now that if a site is using object caching, a transient may not survive the eviction algorithm. Thoughts?

Reply:
> Although it occurs to me now that if a site is using object caching, a transient may not survive the eviction algorithm. Thoughts?

I'd expect that possibility applies to any transient. I haven't seen any code that attempts to mitigate it elsewhere.

Reply:
> haven't seen any code that attempts to mitigate it elsewhere

@rrennick Simplest fix is to use an option instead of a transient. Thoughts?

Reply:
@crstauf I would expect that if eviction were a significant problem then developers would use them as little as possible. If a site had an issue with eviction then it would probably manifest itself in multiple places. 

The advantage of transients is that in a non-persistent cache scenario they are only loaded when needed. In 99% of installs this transient would rarely be accessed.

Reply:
@rrennick Agreed. I only know this because I previously managed an eCommerce site that had issues with object cache evictions.

Is there any opposition against using an option that is not autoloaded?

Reply:
> Is there any opposition against using an option that is not autoloaded?

A transient is an option that's not autoloaded and WP will load a transient from the DB if it's not in cache. I would leave it as a transient since the expiry can be customized.

Reply:
> leave it as a transient

Okay; resolving this conversation.